components/api_version/src/api_v1ttl.rs

```diff
@@ -8,6 +8,7 @@ use super::*;
 
 impl APIVersion for APIV1TTL {
     const TAG: ApiVersion = ApiVersion::V1ttl;
+    const CLIENT_TAG: ApiVersion = ApiVersion::V1;
```

Comment:
what is it used for? why is different from `TAG`

Reply:
`V1ttl` actually stands for `tikv-server` configurated as `storage.enable-ttl = true` (and without setting `api-version = 2`).
To be backward compatible, TiKV with api version `V1ttl` should accept old clients' requests with `V1`. 
Please also see [here](https://github.com/pingcap/kvproto/blob/f7a7c8ccda74ae17acdde7b47cf829ac6f5c2a52/proto/kvrpcpb.proto#L739).

`CLIENT_TAG` is mainly used for generating test cases with expected client request, e.g, [`storage:: test_raw_delete_range`](https://github.com/pingyu/tikv/blob/284d40abb556ab5ff45cbaa6107a6af2f2f468bc/src/storage/mod.rs#L4493).
<img width="1011" alt="image" src="https://user-images.githubusercontent.com/1907938/162183629-bbb55bf8-9755-4f60-aae8-ffe7448e85ea.png">

Reply:
add a #[test] on top of it

Reply:
It seems that codes with #[cfg(test)] can not be exported from crate. And I find this [issue](https://github.com/rust-lang/cargo/issues/8379).

Codes [here](https://github.com/tikv/tikv/pull/12324/files#diff-cfb77164c8825bd898bd32f0ced6b6e1c8564490417a5949e8d9ecf08b9b0341R7):
```rust
impl APIVersion for APIV1 {
    const TAG: ApiVersion = ApiVersion::V1;
    #[cfg(test)]
    const CLIENT_TAG: ApiVersion = ApiVersion::V1;
    const IS_TTL_ENABLED: bool = false;
    ...
```

CI error ([link](https://ci.pingcap.net/blue/organizations/jenkins/tikv-ghpr-clippy-linux-arm64/detail/tikv-ghpr-clippy-linux-arm64/924/pipeline/19/)):
```bash
[2022-04-07T13:10:39.903Z] error[E0599]: no associated item named `CLIENT_TAG` found for type parameter `Api` in the current scope
[2022-04-07T13:10:39.903Z]     --> src/storage/mod.rs:4493:31
[2022-04-07T13:10:39.903Z]      |
[2022-04-07T13:10:39.903Z] 4493 |             api_version: Api::CLIENT_TAG,
[2022-04-07T13:10:39.903Z]      |                               ^^^^^^^^^^ associated item not found in `Api`
[2022-04-07T13:10:39.903Z] 
```

Reply:
I remove `#[cfg(test)]` and add comment that it is "for test only". PTAL~

Reply:
Some modules use `testexport" features for cases like this.

Reply:
I see... Let me try.

Reply:
Done. PTAL~